### DISCUSSION TITLE
Adding Fourier noise to a material property

### DISCUSSION THREAD

**Discussion by vermaprk** [idaholab/moose#19523] [Q&A Modules: General] [answered]
I have material property 'c'. I want another material property 'c_true' which has a Fourier noise in material property 'c'.
I have already defined the function Fourier noise. But how do I add it to my material property c_true?

**Comment by GiudGiud** [comment, marked as answer]
you can use a ParsedMaterial to define c_sum = c + c_noise
https://mooseframework.inl.gov/source/materials/ParsedMaterial.html

**Reply by vermaprk** [reply]
Yes, I tried doing that but it makes the calculations very slow. Is there any efficient way to add Fourier noise or any other noise (https://mooseframework.inl.gov/moose/source/functions/FourierNoise.html) to directly operate on material  property 'c' and give new material property c_true ?

**Reply by GiudGiud** [reply]
I think you'll have to code your own material if you want to make it fast
